lib/python/qmk/info.py

```diff
@@ -70,7 +70,10 @@ def info_json(keyboard):
     info_data = _matrix_size(info_data)
 
     # Merge in data from <keyboard.c>
-    info_data = _extract_led_config(info_data, str(keyboard))
+    try:
+        info_data = _extract_led_config(info_data, str(keyboard))
+    except KeyError as e:
+        _log_error(info_data, 'Unable to extract LED config!')
```

Comment:
Wont this stop the error propagation and actually mask the sort of error?

Reply:
I got this on my test 4x4 keyboard folder for two different reasons - I had a rules.mk in the root which made it a buildable keyboard, however the matrix pin declarations are all in individual rev folder info.jsons for the various dev boards I have.
![image](https://user-images.githubusercontent.com/4781841/200740671-1e997c92-8828-4364-8e3d-05d897d9c316.png)

Also, I accidentally had two `matrix_pins.cols` instead of cols and rows in the Teensy 3.5 info.json.
So, user error I guess, but it prevented `qmk generate-api` from going any further. I'm thinking the latter issue should be caught earlier.

Reply:
#18108 should help with the duplicate cols case